tests/helics/core/ZeromqSSCore-tests.cpp

```diff
@@ -0,0 +1,382 @@
+    comm.disconnect ();
+    core = nullptr;
+    msgs.clear();
+    helics::CoreFactory::cleanUpCores (100ms);
+}
+
+/** test case checks default values and makes sure they all mesh together
+also tests the automatic port determination for cores
+*/
+BOOST_AUTO_TEST_CASE (zmqSSCore_core_broker_default_test)
+{
+    std::string initializationString = "-f 1";
+
+    auto broker = helics::BrokerFactory::create (helics::core_type::ZMQ_SS, initializationString);
+
+    auto core = helics::CoreFactory::create (helics::core_type::ZMQ_SS, initializationString);
+    bool connected = broker->isConnected ();
+    BOOST_CHECK (connected);
+    connected = core->connect ();
+    BOOST_CHECK (connected);
+
+    core->disconnect ();
+
+    BOOST_CHECK (!core->isConnected ());
+    broker->disconnect ();
+    BOOST_CHECK (!broker->isConnected ());
+    helics::CoreFactory::cleanUpCores (200ms);
+    helics::BrokerFactory::cleanUpBrokers (200ms);
+}
+
+class FedTest
+{
+  public:
+    helics::Time deltaTime = helics::Time (10, timeUnits::ns);  // sampling rate
+    helics::Time finalTime = helics::Time (200, timeUnits::ns);  // final time
+  private:
+    std::unique_ptr<helics::ValueFederate> vFed;
+    helics::Publication pub;
+    helics::Input sub;
+    std::string fed_name;
+
+    int pub_index = 0;
+    int sub_index = 0;
+    bool initialized = false;
+
+  public:
+    FedTest () = default;
+
+    void run ()
+    {
+        if (!initialized)
+        {
+            throw ("must initialize first");
+        }
+        vFed->enterInitializingMode ();
+        vFed->enterExecutingMode ();
+        mainLoop ();
+    };
+    void initialize (const std::string &coreName, int p_index, int s_index)
+    {
+        fed_name = "fedtest_" + std::to_string (p_index);
+        pub_index = p_index;
+        sub_index = s_index;
+        helics::FederateInfo fi;
+        fi.coreName = coreName;
+        vFed = std::make_unique<helics::ValueFederate> (fed_name, fi);
+        pub = vFed->registerPublicationIndexed<std::string> ("fedrx", pub_index);
+        sub = vFed->registerSubscriptionIndexed ("fedrx", pub_index);
+        initialized = true;
+    }
+
+    void mainLoop ()
+    {
+        auto nextTime = deltaTime;
+        std::string txstring (fed_name);
+        while (nextTime < finalTime)
+        {
+            nextTime = vFed->requestTime (nextTime + deltaTime);
+            vFed->publish (pub, txstring);
+
+            if (vFed->isUpdated (sub))
+            {
+            	if (vFed->isUpdated(sub)) {
+            		// get the latest value for the subscription
+                    auto &nstring = vFed->getString (sub);
+                    if (nstring != txstring)
+                    {
+                        std::cout << "incorrect string\n";
+                        break;
+                    }
+            	}
+
+            }
+        }
+        vFed->finalize ();
+    }
+};
+
+BOOST_AUTO_TEST_CASE (zmqSSMultiCoreInitialization_test)
```

Comment:
I think my only concern is that this test case doesn't have any tests/checks in it.  It also uses a lot of sleeps which is a little concerning?    There should be something to indicate it has passed or failed.